### PR TITLE
Added gcloud plugin and nbactions.xml

### LINKED ISSUE #5: add gcloud plugin to guestbook sample

### PR DESCRIPTION
**Pull request by lesv**
1. Update all files to use both app engine & gcloud plugins
2. move nbactions to non-filtered.
3. Add back nbactions everywhere
4. add .gitignore everywhere
5. fix some linefeed eol’s in places.

fixes #5 #7

### CODE REVIEW

**Comment by lesv** [comment]
Deals w/ #7 and much more.

File: appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 1]
```diff
@@ -27,15 +27,16 @@
         <include>**/*.launch</include>
       </includes>
     </fileSet>
-    <fileSet encoding="UTF-8">
+    <fileSet filtered="true" encoding="UTF-8">
       <directory></directory>
       <includes>
-        <include>nbactions.xml</include>
+        <include>pom.xml</include>
```
Comment (tmatsuo) [inline appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: Hi, I don't think you need to have pom.xml explicitly. It worked properly without having this entry right?

File: endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml [hunk 2]
```diff
@@ -1,143 +1,155 @@
...
-</project>
+    <modelVersion>4.0.0</modelVersion>
+    <packaging>war</packaging>
+    <version>${version}</version>
+
+    <groupId>${groupId}</groupId>
+    <artifactId>${artifactId}</artifactId>
+
+    <properties>
+        <app.id>${application-id}</app.id>
```
Comment (tmatsuo) [inline endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:11]: What's the motivation of having app.id and app.verison here?

With the gcloud migration, I think we will remove app.id and app.version from appengine-web.xml. If it's true, after removing those lines from appengine-web.xml, app.id and app.version in pom.xml will become just garbage.

Sorry if I miss something.

File: guestbook-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 3]
```diff
@@ -35,10 +35,10 @@
         <include>**/*.launch</include>
       </includes>
     </fileSet>
-    <fileSet encoding="UTF-8">
+    <fileSet filtered="true" encoding="UTF-8">
       <directory></directory>
       <includes>
-        <include>nbactions.xml</include>
+        <include>pom.xml</include>
```
Comment (tmatsuo) [inline guestbook-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: ditto do you really need to add pom.xml?

File: endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml [hunk 4]
```diff
@@ -33,14 +33,16 @@
...
+        <include>pom.xml</include>
       </includes>
     </fileSet>
     <fileSet encoding="UTF-8">
       <directory></directory>
       <includes>
         <include>LICENSE</include>
+        <include>README.md</include>
+        <include>nbactions.xml</include>
+        <include>.gitignore</include>
```
Comment (tmatsuo) [inline endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: It's not true everyone uses git.

**Comment by tmatsuo** [comment]
Overall, looks good.
1. I'd rather not add .gitignore
2. I'd rather not add pom.xml in the metadata unless there is a good reason to add and you're 100% sure it's not breaking anything since we have been good without having it for over a year.
3. app.id and app.version. If we will remove app.id and app.version from appengine-web.xml sometime in the future, please don't include them in pom.xml. If we keep them in appengine-web.xml, that's fine.

File: appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 1 above)
Comment (lesv) [inline appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: It did, but once we remove nbactions.xml, the lines look weird.  This made things explicit.

File: endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (lesv) [inline endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:11]: It gives a high level place to change things.  Yep - once gcloud migration fully happens, this should go away.  But it hasn't happened yet.

File: endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 4 above)
Comment (lesv) [inline endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: It was in one of the templates, so I propagated it everywhere.  I was trying to make them the same on non-substantive issues.

Comment (lesv) [inline endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: If you want .gitignore pulled from everywhere, I'm actually Ok with that.

File: endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (lesv) [inline endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:11]: Putting them in the pom let's us deal with passing metadata for gcloud later.

File: endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 4 above)
Comment (tmatsuo) [inline endpoints-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: Yeah, let's remove .gitignore.

File: appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml (hunk 1 above)
Comment (tmatsuo) [inline appengine-skeleton-archetype/src/main/resources/META-INF/maven/archetype-metadata.xml]: In that case, can you just remove the fileSet?

File: endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml (hunk 2 above)
Comment (tmatsuo) [inline endpoints-skeleton-archetype/src/main/resources/archetype-resources/pom.xml:11]: Ok, if we plan to use the values in pom.xml, that's great.

**Comment by lesv** [comment]
PTAL @tmatsuo

**Comment by tmatsuo** [comment]
LGTM
Thanks!

**Pull request merged**